src/Compilers/CSharp/Portable/Lowering/AsyncRewriter/AsyncMethodBuilderMemberCollection.cs

```diff
@@ -422,6 +424,15 @@ private static NamedTypeSymbol ValidateBuilderType(SyntheticBoundNodeFactory F,
                     !property.IsStatic &&
                     (property.ParameterCount == 0))
                 {
+                    if (property.Type != returnType)
```

Comment:
If our goal is to ensure that we can pass a value from one API to another, none of the mismatches matter. I can see that we might want to require dynamic/tuple names to match, we can invent such language rule. Custom modifiers, however, should be ignored, unless CLR requires them to match. If that is the case, why do we ignore custom modifiers on the property itself? Feels inconsistent to me.